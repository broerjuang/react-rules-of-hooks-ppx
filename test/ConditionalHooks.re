let useMouseHook = () => ();

[@react.component]
let make = (~randomProp) => {
  if (randomProp === "state") {
    useMouseHook();
  };

  <div />;
};
