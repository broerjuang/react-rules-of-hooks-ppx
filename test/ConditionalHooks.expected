[1mFile "ConditionalHooks.ml", line 3, characters 32-47[0m:
3 |   if randomProp == "state" then useMouseHook ();
                                    [1;31m^^^^^^^^^^^^^^^[0m
[1;31mError[0m: Hooks can't be inside conditionals, neither loops.
